bin/java

```diff
@@ -14,6 +14,40 @@ install_java_with_overlay() {
   local buildDir="${1}"
   local cacheDir="${2:-$(mktemp -d)}"
   if [ ! -f "${buildDir}/.jdk/bin/java" ]; then
+
+    detectedVersion=$(_get_system_property "${buildDir}/system.properties" "java.runtime.version")
+    if [ -z "${detectedVersion}" ]; then
+      if [ "${STACK}" == "heroku-24" ]; then
+        warning "No OpenJDK version specified
+Your application does not explicitly specify an OpenJDK version. The latest
+long-term support (LTS) version will be installed. This currently is OpenJDK ${DEFAULT_JDK_VERSION}.
+
+This default version will change when a new LTS version is released. Your
+application might fail to build with the new version. We recommend explicitly
+setting the required OpenJDK version for your application.
+
+To set the OpenJDK version, add or edit the system.properties file in the root
+directory of your application to contain:
+
+java.runtime.version = ${DEFAULT_JDK_VERSION}
+"
+      else
+        warning "No OpenJDK version specified
+Your application does not explicitly specify an OpenJDK version.
+OpenJDK ${DEFAULT_JDK_VERSION} will be installed.
+
+This default version will change at some point. Your
```

Comment:
(optional) The wrapping on this line (and some others) looks a little strange when viewed as the preview in the PR description, since the line lengths vary so much from line to line.

Reply:
Adjusted in [57dad16](https://github.com/heroku/heroku-buildpack-jvm-common/pull/301/commits/57dad16ded039b4aaedd810f02db5fa9c27bbcbe).